compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -171,6 +171,57 @@ impl<K: DepKind> DepGraph<K> {
         K::with_deps(None, op)
     }
 
+    /// Used to wrap the deserialization of a query result from disk,
+    /// This method enforces that no new `DepNodes` are created during
+    /// query result deserialization.
+    ///
+    /// Enforcing this makes the query dep graph simpler - all nodes
+    /// must be created during the query execution, and should be
+    /// created from inside the 'body' of a query (the implementation
+    /// provided by a particular compiler crate).
+    ///
+    /// Consider the case of three queries `A`, `B`, and `C`, where
+    /// `A` invokes `B` and `B` invokes `C`:
+    ///
+    /// `A -> B -> C`
+    ///
+    /// Suppose that decoding the result of query `B` required invoking
+    /// a query `D`. If we did not create a fresh `TaskDeps` when
```

Comment:
I would prefer if we changed that to

> Suppose that decoding the result of query `B` required re-computing query `C`.

That would be OK in principle since it would be compatible with the dep-graph we already created. But creating any edge away from `B` will never be legal (not even if we allowed query invocation during cache loading) because we already allocated `B`'s dep-node during red-green marking. The current wording makes it sound like it would be OK for a "new" query to be called during decoding.

Reply:
My thinking was to explain the process for an arbitrary new node, and then explain that this can only work if the node already exists. However, I'm fine with changing the wording.